rfcs/text/remove-support-for-uncontrolled-inputs.md

```diff
@@ -0,0 +1,224 @@
+
+# Adoption strategy
+
+As mentioned, this proposed work will be implemented as a `BREAKING CHANGE`. In terms of how consuming projects will upgrade Carbon, we will need to document the upgrade process and link to it in the release notes. If a project has used an input as uncontrolled, they will need to update their codebase to make this controlled instead. This will involve making sure that the value is stored using React's [useState](https://reactjs.org/docs/hooks-state.html) hook to retain the value and making sure this is managed using an `onChange` function. We may have to provide additional support in this scenario to provide examples of how they can achieve the same functionality using this approach.
+
+We also already provide various codemods in our [Carbon Codemod](https://github.com/Sage/carbon-codemod) repo that could be used by our consumers to assist with the move to a controlled component. Examples of this are:
+- A [rename-prop](https://github.com/Sage/carbon-codemod/tree/master/transforms/rename-prop) codemod that consumers could use to change a `defaultValue` value prop to `value`.
+- A [add-prop](https://github.com/Sage/carbon-codemod/tree/master/transforms/add-prop) codemod that could be used to add to a `value` prop.
+- A [remove-prop](https://github.com/Sage/carbon-codemod/tree/master/transforms/remove-prop) codemod that could be used to remove a `defaultValue` prop.
+- A [replace-prop-value](https://github.com/Sage/carbon-codemod/tree/master/transforms/replace-prop-value) codemod that changes the value of a prop.
+
+There is also a lot of support for writing controlled components.
+
+# How we teach this
+
+This RFC will be communicated in the Sage `#carbon` and `#carbon-announcement` slack channels, where we will ask our consumers to feedback on the RFC. Beyond that, it will be visible in the release notes when consumers upgrade. This RFC document will reside in the repository for future reference.
+
+We will be able to support consumers of Carbon in the #carbon slack channel too. 
+
+# Documentation
+
+As part of this work we will ensure that Storybook examples are updated where any uncontrolled behaviours are demonstrated.
+
+# Further Considerations
+
+If/when we introduce a `File Input` component in to Carbon, this will have to be an uncontrolled component. This is because `<input type="file" />` can only be uncontrolled as [the value can only be specified by the user and not be done programmatically](https://reactjs.org/docs/uncontrolled-components.html#the-file-input-tag). As a file input only works uncontrolled, we would not need to maintain any controlled behaviour.
```

Comment:
```suggestion
If/when we introduce a `FileInput` component into Carbon, this will have to be an uncontrolled component. This is because `<input type="file" />` can only be uncontrolled as [the value can only be specified by the user and not be done programmatically](https://reactjs.org/docs/uncontrolled-components.html#the-file-input-tag). As a file input only works uncontrolled, we would not need to maintain any controlled behaviour.
```